Switch to using gratia

gavinsimpson opened the pull request:
This PR implements all the changes required to move to using gratia for the plots in the paper. (Issue #37 )

It looks like this isn't going to merge cleanly because of changes others have made already, and I can't compile locally in the branch because there is a problem in generating one of the early figures (the basis functions one)

Can you take a quick look at this and see what needs to be done to merge this? Once it's in I'll take a final look over the outputs and insure everything is fixed.

I plan to release gratia 0.2-0 to CRAN shortly but any version from >= 0.1-9 is OK now. Barring issue 0.2-0 should be on CRAN as soon as CRAN can process it (I'll submit it tomorrow once I've run a few checks against R-devel).

eric-pedersen commented:
Hi @gavinsimpson, I've merged the main branch in to this, so it should  be good to go, pending a final check from you. I've tweaked the build file too, so I think it should compile more seamlessly now (although I keep running into different build problems on different laptops; it just seems like it's really difficult to create a build process that works on all platforms with this many moving parts).

gavinsimpson commented:
Thanks @eric-pedersen I'll take a look locally. One issue is that I think there is an issue in the basis function code - I was running code locally and couldn't get that chunk to run. As mentioned in #44 I think we nee to delete the cache once we're at a final version and see if all the chunks build. But let's tackle that once this is merged and we are happy we've addressed all the reviewer comments.

eric-pedersen commented:
Okay, I think I figured out what the issue might have been; I had declared `figures/alternate_models.png` as a target in the makefile, which means that if that file is absent (as it should be in a clean run of the build), the makefile raises an error.  I've tried changing the target to `code/alternate_models.R` and am now compiling the paper from a clean start. If that works (should know in ~30 min to an hour) I'll push the fixed makefile to this branch. If it still isn't working for you, let me know what error you're getting and I'll try to recreate.

eric-pedersen commented:
@gavinsimpson, the version I just pushed to Github compiles from a clean cache and figure folder for me; try it now.
